server/interfaces/interfaces.go

```diff
@@ -828,6 +828,8 @@ type FileCache interface {
 	Read(ctx context.Context, node *repb.FileNode) ([]byte, error)
 	Write(ctx context.Context, node *repb.FileNode, b []byte) (n int, err error)
 
+	VerifiedWriter(ctx context.Context, node *repb.FileNode, digestFunction repb.DigestFunction_Value) (CommittedWriteCloser, error)
```

Comment:
nit: I think it'd be OK to just call this Writer and leave a comment above explaining that the file is only added to the filecache if the written bytes match the nodehash.

Reply:
Done. The original motivation was to differentiate it from `Write` which intentionally does not verify the contents.